presto-spi/src/main/java/com/facebook/presto/spi/type/DoubleType.java

```diff
@@ -68,8 +70,8 @@ public long hash(Block block, int position)
     @Override
     public int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
     {
-        double leftValue = leftBlock.getDouble(leftPosition, 0);
-        double rightValue = rightBlock.getDouble(rightPosition, 0);
+        double leftValue = longBitsToDouble(leftBlock.getLong(leftPosition, 0));
```

Comment:
Instead of adding this code everywhere, what do you think about adding getDouble to the Block interface with a default method that does the longBitsToDouble operation?